components/org.wso2.carbon.identity.context.rewrite.valve/src/main/java/org/wso2/carbon/identity/context/rewrite/valve/TenantContextRewriteValve.java

```diff
@@ -72,12 +75,18 @@ public void invoke(Request request, Response response) throws IOException, Servl
 
         //Get the rewrite contexts and check whether request URI contains any of rewrite contains.
         for (RewriteContext context : contextsToRewrite) {
-            Pattern pattern = context.getPattern();
-            if (pattern.matcher(requestURI).find() || pattern.matcher(requestURI + "/").find()) {
+            Pattern patternTenant = context.getTenantContextPattern();
+            Pattern patternSuperTenant = context.getBaseContextPattern();
+            if (patternTenant.matcher(requestURI).find() || patternTenant.matcher(requestURI + "/").find()) {
                 isContextRewrite = true;
                 isWebApp = context.isWebApp();
                 contextToForward = context.getContext();
                 break;
+            } else if (patternSuperTenant.matcher(requestURI).find() || patternSuperTenant.matcher(requestURI + "/")
+                    .find()) {
+                String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
```

Comment:
Can't we directly set the super tenant domain name?

Reply:
That will add some readability. But, tenant is originally being set in a lower level valve engaged from platform.
Do you think that this would lead to any confusions or implications

Reply:
I don't see any issue as long as we can guarantee that the tenant domain is set from a lower level. But since we know the tenant is 'carbon.super' I felt it would be easier and much readable if we set the value directly here.

Reply:
+1 let's set it